pysqldb3/geopackage.py

```diff
@@ -75,8 +83,32 @@ def write_gpkg(self, tbl_name = None, print_cmd=False):
         if not self.path:
             self.path = file_loc('folder')
 
-        if not self.table:
-            self.table = tbl_name
+        if not self.gpkg_tbl:
+            self.gpkg_tbl = self.table
+
+        # overwrite vs update vs an issue has arisen
+        if overwrite:
+            # if explict overwrite, then create command line as directed
+            _update = ''
+            _overwrite = '-overwrite'
+        
+        elif not overwrite and not os.path.isfile(os.path.join(self.path, self.gpkg_name)):
+            # this geopackage does not exist so create as if new
+            _update = ''
+            _overwrite = ''
+        
+        elif not overwrite and os.path.isfile(os.path.join(self.path, self.gpkg_name)): # check if the geopackage already exists
+            
+            try:
+                exists_cmd = f'ogrinfo {os.path.join(self.path, self.gpkg_name)}'
+                ogr_response = subprocess.check_output(exists_cmd, stderr=subprocess.STDOUT)
+                table_exists = re.findall(f"{self.gpkg_tbl}", str(ogr_response)) # only allows tables names with underscores, numbers, and letters
+                assert len(table_exists) == 0, "The table name to be exported already exists in the geopackage. Either change to Overwrite = True or check the name of the table to be copied."
+
+            finally:
```

Comment:
The `try` statement is to see if my command returns a valid message, and if it does, it counts how many items are in the list.  And if it doesn't, then it creates an empty list and counts it as 0, rather than giving me an error. Maybe I don't need to use `finally` (?) but I didn't think it should be an `else:` because i want to come up with a number at the end.

Reply:
we do need to catch the error, because i can't count the number of entries in a list if the output isn't a list. as a specific example, i run a windows command that tells me what tables are in the geopackage. but if the geopackage isn't working or there's an error, i won't get a list of the tables and instead, i'll get an error message; in that case, i will just manually set the table list to an empty list. if the geopackage exists, i should return an output that i can parse for the table count. but if it doesn't exist, there is nothing to parse. that is why i have the try statement to catch the error.

Reply:
what happens if you dont have overwrite or update flags and you try to write to gpkg with a table that already exists? I assume it throws an error, why not return that error with a note about using the overwrite or update flags to the user - it should fail if you try it and there is a conflict and you havent explicitly stated overwrite or update

Reply:
i dont see it in the exception